src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/LosAlamosFFTWriter.cpp

```diff
@@ -0,0 +1,85 @@
+#include "LosAlamosFFTWriter.hpp"
+
+#include "complex/Common/Constants.hpp"
+#include "complex/DataStructure/DataArray.hpp"
+#include "complex/DataStructure/Geometry/ImageGeom.hpp"
+#include "complex/Utilities/FilterUtilities.hpp"
+
+#include <fstream>
+
+namespace fs = std::filesystem;
+using namespace complex;
+
+namespace
+{
+using ull = unsigned long long int;
+}
+
+// -----------------------------------------------------------------------------
+LosAlamosFFTWriter::LosAlamosFFTWriter(DataStructure& dataStructure, const IFilter::MessageHandler& mesgHandler, const std::atomic_bool& shouldCancel, LosAlamosFFTWriterInputValues* inputValues)
+: m_DataStructure(dataStructure)
+, m_InputValues(inputValues)
+, m_ShouldCancel(shouldCancel)
+, m_MessageHandler(mesgHandler)
+{
+}
+
+// -----------------------------------------------------------------------------
+LosAlamosFFTWriter::~LosAlamosFFTWriter() noexcept = default;
+
+// -----------------------------------------------------------------------------
+const std::atomic_bool& LosAlamosFFTWriter::getCancel()
+{
+  return m_ShouldCancel;
+}
+
+// -----------------------------------------------------------------------------
+Result<> LosAlamosFFTWriter::operator()()
+{
+  /**
+   * Header print function was unimplemented in original filter so it was omitted here and condensed to just the writeFile() function
+   */
+
+  // Make sure any directory path is also available as the user may have just typed
+  // in a path without actually creating the full path
+  Result<> createDirectoriesResult = complex::CreateOutputDirectories(m_InputValues->OutputFile.parent_path());
+  if(createDirectoriesResult.invalid())
+  {
+    return createDirectoriesResult;
+  }
+
+  std::ofstream file = std::ofstream(m_InputValues->OutputFile, std::ios_base::out);
+  if(!file.is_open())
+  {
+    return MakeErrorResult(-73450, fmt::format("Error creating and opening output file at path: {}", m_InputValues->OutputFile.string()));
+  }
+
+  SizeVec3 dims = m_DataStructure.getDataAs<ImageGeom>(m_InputValues->ImageGeomPath)->getDimensions();
+
+  auto cellEulerAngles = m_DataStructure.getDataRefAs<Float32Array>(m_InputValues->CellEulerAnglesArrayPath);
+  auto cellPhases = m_DataStructure.getDataRefAs<Int32Array>(m_InputValues->CellPhasesArrayPath);
+  auto featureIds = m_DataStructure.getDataRefAs<Int32Array>(m_InputValues->FeatureIdsArrayPath);
```

Comment:
```suggestion
  auto& cellEulerAngles = m_DataStructure.getDataRefAs<Float32Array>(m_InputValues->CellEulerAnglesArrayPath);
  auto& cellPhases = m_DataStructure.getDataRefAs<Int32Array>(m_InputValues->CellPhasesArrayPath);
  auto& featureIds = m_DataStructure.getDataRefAs<Int32Array>(m_InputValues->FeatureIdsArrayPath);
```
I'd also recommend getting calling `getDataStoreRef()` and using that when referencing indices in loops instead of the array class. It's a small change, but the constant lookup checks can build up.

Reply:
Also on the original point, use `auto` when you want a local copy and `auto&` when you want to share a reference. Unless you think the DataStructure is going out of scope while this operation is going on, we don't need extra copies of each array.

It will still work since DataArray uses shared_ptr, there's no need for the extra memory usage.